### PR TITLE
Introduce 'offers' subresource to v2 API

### PR DESCRIPTION
**Pull request by kjdchapman**
This PR introduces an index view of the 'offers' subresource, which helps to describe the current status of an application.

This will be expanded to include POSTing to this endpoint to make offers to candidates - the intent is for this to be more expressive than v1's `make_offer` and `reject` endpoints.
___

This PR has several linting violations, but they're mainly single/double quote things and would potentially make this change confusing to parse/review. 

As [previously discussed](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/60#issuecomment-488305332), I'd prefer to fix the violations in a future PR that includes TravisCI linting support, so it can't go wrong from there.

### CODE REVIEW

File: app/controllers/api/v1/applications_controller.rb [hunk 1]
```diff
@@ -4,32 +4,32 @@ def index
   end
 
   def show
-    if build_application.present?
```
Comment (heidar) [inline app/controllers/api/v1/applications_controller.rb]: Hmm why the change from `build_`?

Comment (kjdchapman) [reply app/controllers/api/v1/applications_controller.rb]: Oops, this was my mistake when resolving a merge - intended to stick with `build_`! Will fix it.
